server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4540,12 +4542,137 @@ public void syncSegmentsFromRemoteSegmentStore(boolean overrideLocal, boolean re
+        RemoteSegmentStoreDirectory remoteSegmentStoreDirectory = null;
+        if (remoteStore != null) {
+            remoteSegmentStoreDirectory = getRemoteSegmentDirectoryForShard();
+            remoteSegmentStoreDirectory.init();
+            remoteStore.incRef();
+        }
+        sourceRemoteSegmentDirectory.init();
+        Map<String, RemoteSegmentStoreDirectory.UploadedSegmentMetadata> uploadedSegments = sourceRemoteSegmentDirectory
+            .getSegmentsUploadedToRemoteStore(primaryTerm, commitGeneration);
+        final Directory storeDirectory = store.directory();
+        store.incRef();
+
+        try {
+            String segmentInfosSnapshotFilename = copySegmentFilesAndGetSegmentInfosFileName(storeDirectory,
+                sourceRemoteSegmentDirectory, remoteSegmentStoreDirectory, uploadedSegments, true,
+                overrideLocal);
+            if (segmentInfosSnapshotFilename != null) {
+                if (remoteSegmentStoreDirectory != null) {
+                    remoteSegmentStoreDirectory.uploadMetadata(uploadedSegments.keySet(), storeDirectory, primaryTerm, commitGeneration);
+                }
+                try (
+                    ChecksumIndexInput indexInput = new BufferedChecksumIndexInput(
+                        storeDirectory.openInput(segmentInfosSnapshotFilename, IOContext.DEFAULT)
+                    )
+                ) {
+                    SegmentInfos infosSnapshot = SegmentInfos.readCommit(store.directory(), indexInput, commitGeneration);
+                    long processedLocalCheckpoint = Long.parseLong(infosSnapshot.getUserData().get(LOCAL_CHECKPOINT_KEY));
+                    if (remoteStore != null) {
+                        store.commitSegmentInfos(infosSnapshot, processedLocalCheckpoint, processedLocalCheckpoint);
+                    }
+                    else {
+                        store.directory().sync(infosSnapshot.files(true));
+                        store.directory().syncMetaData();
+                    }
+                }
+            }
+        } catch (IOException e) {
+            throw new IndexShardRecoveryException(shardId, "Exception while copying segment files from remote segment store", e);
+        } finally {
+            store.decRef();
+            if (remoteStore!=null) {
+                remoteStore.incRef();
+            }
+        }
+    }
+
+    private String copySegmentFilesAndGetSegmentInfosFileName(Directory storeDirectory,
+                                                           RemoteSegmentStoreDirectory sourceRemoteDirectory,
+                                                           RemoteSegmentStoreDirectory targetRemoteDirectory,
+                                                           Map<String, RemoteSegmentStoreDirectory.UploadedSegmentMetadata> uploadedSegments,
+                                                           boolean syncTillCommitPoint,
```

Comment:
if this is `true` , lets skip for now.

Reply:
this is now being used in `syncSegmentsFromRemoteSegmentStore` method as well, where we are passing it false.

Reply:
yep.. this was not needed, removed it